### PR TITLE
Upload post func

### LINKED ISSUE #1610: Post Not Being Created in Database after Posting Something
**Describe the bug**
When we create a post using the ```Add``` button in the app, it does nothing, and shows ```Post is uploaded```, but, inside the database, we don't have any post created.

**To Reproduce**

1. Go to Talawa-App
2. Create a new post
3. Check database for all post documents
4. No document will be there for recently created post

**Expected behavior**
There should be a post created.

**Actual behavior**
No post is being created.

**Screenshots**
image
image

**Additional details**
No

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Kushalrock**
What kind of change does this PR introduce?
Upload post functionality was just a placeholder and wasn't actually uploading anything to the database. In fact there was no network call whatsoever. Fixed that by making a custom query. Added a title field in add post page.

Issue Number:
Closes #1610 

Have you read the [contributing guide]
Yes

Summary
1. Upload post functionality was just a placeholder and wasn't actually uploading anything to the database. In fact, there was no network call whatsoever.
2. Added a title text field because it was something that was being checked in the backend for uploading post
3. Made a custom query for uploading post to the database

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing test, and read about the new comment inter in the CONTRIBUTING.md file

**Comment by Kushalrock** [comment]
Yeah @palisadoes I thought I did that but this linting check many other things too. No problem I resolved it.

**Comment by Kushalrock** [comment]
@palisadoes It will take some time to develop teh tests for it. Even it should be a separate issue because we have to write custom function sin test helpers and such for mocking upload functionality.

**Comment by palisadoes** [comment]
1. Writing tests should not be neither difficult not time consuming for this. It's a single upload feature. 
1. We need quality code. Talawa has been unusable due to past bugs. We cannot get back to that.

Please take the time to do this.

**Comment by Kushalrock** [comment]
Nope @palisadoes the thing is there is no foundation in the test helpers thingy because before I added network stuff there was nothing for which mock function had to be written. So, the person who will be writing tests will also have to write that foundation in test helpers mock and then extend it in test helpers. And after that write any other test.

**Comment by palisadoes** [comment]
OK. Please create the issues for those now so we don't forget.

I'll request approved in the meantime

File: test/view_model_tests/after_auth_view_model_tests/add_post_view_model_test.dart [hunk 1]
```diff
@@ -67,25 +66,25 @@ void main() {
...
+    //   expect(model.imageFile, null);
+    //   verify(notifyListenerCallback());
+    //   expect(model.controller.text, "");
+    //   verify(
+    //     navigationService.showTalawaErrorSnackBar(
+    //       "Post is uploaded",
+    //       MessageType.info,
+    //     ),
+    //   );
+    // });
```
Comment (noman2002) [inline test/view_model_tests/after_auth_view_model_tests/add_post_view_model_test.dart:88]: Please remove it if its not needed later.

**Review by noman2002 — DISMISSED** [review]
Rest everything looks good

File: test/view_model_tests/after_auth_view_model_tests/add_post_view_model_test.dart (hunk 1 above)
Comment (palisadoes) [reply test/view_model_tests/after_auth_view_model_tests/add_post_view_model_test.dart:88]: @Kushalrock Please make the requested updates

Comment (Kushalrock) [reply test/view_model_tests/after_auth_view_model_tests/add_post_view_model_test.dart]: @palisadoes @noman2002 I have done so you guys can approve it for now. Also, please look at issue #1680 and edit it to the test format. But yeah please don't add the good first issue label as this is not something to tackle for someone with beginner flutter experience instead someone who knows a bit more about codebase. I may take up the issue

**Review by palisadoes — APPROVED** [review]

**Review by noman2002 — APPROVED** [review]

**Comment by Kushalrock** [comment]
@Ayush0Chaudhary Nope post fetching is another different issue.

**Pull request merged**